### DISCUSSION TITLE
Creating Complex Filters

### DISCUSSION THREAD

**Discussion by Ceequester** [stashapp/stash#5524] [Ideas]
So, I don't have time to fully rabbit-hole my way into a design for this right now but one thing that would be greatly appreciated is a way to convey complex filters through the current filter system. I would be satisfied with just a "Add your own GraphQL filter" that would be injected in instead of building a filter but eventually it would be nice if I could convey the same level of detailed logic with the UI as I can with the GraphQL filter. For an example of something that I don't believe is possible with the current logic here's a query I would like to be able to use as a filter:
```graphql
{
  findScenes(scene_filter: {
      AND:{
        OR: {
          stash_id_endpoint: {
            endpoint: "https://theporndb.net/graphql",
            modifier: IS_NULL
          }
          tags: {
          modifier: INCLUDES_ALL
            excludes: [1550]
          }
        }
        stash_id_endpoint: {
          endpoint: "https://stashdb.org/graphql",
          modifier: IS_NULL
        }
        tags: {
          modifier: INCLUDES_ALL
          excludes: [1549]
        }
      }
      path: {
        value: "/data/archive/adult/videos/"
        modifier: MATCHES_REGEX
      }
    }) {
    scenes { 
      title
      files {
          path
        }
    }
  }
}
```

**Comment by feederbox826** [comment]
if you urlencode your filter, you can just append it

**Reply by Ceequester** [reply]
Do you have a small example of what you mean? It's not really clear to me what parameter to add it as or how to respect the parser that's being used internally in the UI.
